sdk/java/pulumi/src/main/java/com/pulumi/core/Alias.java

```diff
@@ -44,11 +41,6 @@ public class Alias {
     private final Output<String> parentUrn;
     private final boolean noParent;
 
-    @SuppressWarnings("unused")
-    private Alias() {
-        throw new UnsupportedOperationException("static class");
```

Comment:
this only guarded against construction though reflection, not a concern IMO

If any other constructor is present, Java won't generate the default no-args constructor.